src/server/game/Server/GameClient.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * This file is part of the TrinityCore Project. See AUTHORS file for Copyright information
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef __GAMECLIENT_H
+#define __GAMECLIENT_H
+
+#include "WorldSession.h"
+
+class Player;
+class Unit;
+
+class TC_GAME_API GameClient
+{
+    public:
+        GameClient(WorldSession* sessionToServer);
+
+        void AddAllowedMover(Unit* unit);
+        void RemoveAllowedMover(Unit* unit);
+        bool IsAllowedToMove(Unit* unit) const;
+        bool IsAllowedToMove(ObjectGuid guid) const;
+        void SetMovedUnit(Unit* target, bool allowMove);
+
+        Unit* GetActiveMover() const { return _activeMover; }
+        void SetActiveMover(Unit* activeMover) { _activeMover = activeMover; }
```

Comment:
...or is a "mover" something that is moved _by_ us?

This naming seems really confusing (very similar to the old naming scheme which was replaced in #17819). I think it's extremely likely to cause more issues over time. I'd recommend renaming.

Reply:
The active mover of a game client is the unit that a client is currently moving. The naming actually come from the opcodes `CMSG_SET_ACTIVE_MOVER ` and `CMSG_MOVE_NOT_ACTIVE_MOVER` so I feel like it is sensible.

The client sends these kind of messages when it starts/stops moving a unit (after it has been authorized/unauthorized to do so by the server using `SMSG_CONTROL_UPDATE`.

Reply:
Even if the opcodes are called that, the name is still horribly un-intuitive. Do consider that this logic will be used by folks five years down the line, who won't be able to phone up you or me to explain it to them.

I'd recommend naming them `GetUnitBeingMoved` or something (isn't that what we currently have?)

#17819 was in part caused due to bugs, which had been caused by people assuming "mover" meant what one might intuitively think it means (the unit moving me).

Reply:
What about `activeMovedUnit` ? Multiple opinions would be appreciated. I'm also fine with `activeMover` since a gameClient cannot be "moved" by a unit. Only a unit can be "moved" by a game client. Removing the cyclic reference already defuses a lot of confusion imo.

Reply:
Done. Renamed the field to `activelyMovedUnit`.